on_boot.d/99-udm-le.sh

```diff
@@ -1,10 +1,20 @@
 #!/bin/sh
 
-UDM_LE_PATH='/mnt/data/udm-le'
+#Load Environmnt Variables
+. /mnt/data/udm-le/udm-le.env
+
+
+#If flag is set, the following section will re-apply the existing certificate during boot
+
+if [ "$ENABLE_CAPTIVE" == "yes" ]; then
```

Comment:
So what you're doing is along the lines of what I was thinking but there's a couple of things I'm not sure about here: 

1) I get that if you're using captive portal you don't want it to be broken for X minutes post reboot until the cert was fixed, but depending on when exactly this gets triggered post reboot you're going to be restarting the controller, possibly interrupting important stuff.

2) I was trying to think of some if/else scenario here, because let's say on reboot your cert was expired, this would restart the controller using an expired cert, then 5 minutes later restart it again with a valid one. Like maybe swap these around so the sleep & renew attempt happen then if ENABLE_CAPTIVE do a restart just in case? The 300 second sleep was totally arbitrary, it may very well be something like 120s would be adequate, I just didn't feel like testing it by rebooting my router a bunch.

Reply:
You could also move the java cert import in `udm-le.sh` out of deploy_cert and keep it wrapped in the if, that way even if there wasn't a new cert it would do the import?

Reply:
Yeah I see your point.  The `on_boot.d` script could apply either the new cert (if renewed) or the existing cert if not renewed that way.  That would mean, though, that the cronjob would also re-apply the cert without needing to and restarting unifi-os.

Reply:
How about this:

If you remove the ENABLE_CAPTIVE statement from the `on_boot.d` script totally, but leave it in the `udm-le.sh` script but in the main bit of the script not the deploy_cert() bit and wrapped in an if, it COULD import and restart every time it was run.

Now this would mean the cron restarts the controller every night if ENABLE_CAPTIVE was set, but maybe the on_boot.d script exports a variable that `udm-le.sh` looks for to know if it was invoked at boot time instead of via cron, so it wouldn't end up restarting the cert every night.

Reply:
Alternatively, you could move the logic for the portal stuff in the main script to a function, but bake in some logic that does an md5sum or something on `${UDM_LE_PATH}/lego/certificates/${CERT_NAME}.crt ${UBIOS_CERT_PATH}/unifi-core.crt`. If they don't match, you know you probably just rebooted, and you could copy and do the restart. The trick would be not doing a double restart on a new cert deploy.

Reply:
Take a look at my last commit, I think I've solved the handling of boot workflow and cronjob workflow.

Reply:
Sorry I forgot to remove the unneeded on_boot things and add the new version of the script call, see my latest commit.